tests/Unit/Domain/Test_Orientations.cpp

```diff
@@ -0,0 +1,181 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
```

Comment:
If you've made a copy of `Time` you've already said that the original and the copy are equal, otherwise how could you have made a copy? In order to define "this is a copy of that" you've already had to define that the states are equal.

For abstract base classes the situation is less clear and this is why I said there are good reasons to not have it in some cases. The problem with abstract base classes is that they are an interface, not a value. Whether it makes sense to compare interfaces I think is a completely different discussion.